File: examples/javascript/ollama_client.js
/**
 * Cliente JavaScript para API do Ollama
 * Exemplo prático de integração com IA local
 */

class OllamaClient {
    constructor(baseUrl = 'http://localhost:11434') {
        this.baseUrl = baseUrl;
    }

    /**
     * Lista todos os modelos disponíveis
     */
    async listModels() {
        try {
            const response = await fetch(`${this.baseUrl}/api/tags`);
            return await response.json();
        } catch (error) {
            console.error('Erro ao listar modelos:', error);
            throw error;
        }
    }

    /**
     * Gera resposta para um prompt
     */
    async generate(prompt, model = 'llama2', stream = false, options = {}) {
        try {
            const data = {
                model,
                prompt,
                stream,
                ...options
            };

            const response = await fetch(`${this.baseUrl}/api/generate`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(data)
            });

            return await response.json();
        } catch (error) {
            console.error('Erro ao gerar resposta:', error);
            throw error;
        }
    }

    /**
     * Chat com histórico de mensagens
     */
    async chat(messages, model = 'llama2') {
        try {
            const data = {
                model,
                messages
            };

            const response = await fetch(`${this.baseUrl}/api/chat`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(data)
            });

            return await response.json();
        } catch (error) {
            console.error('Erro no chat:', error);
            throw error;
        }
    }

    /**
     * Baixa um modelo específico
     */
    async pullModel(modelName) {
        try {
            const response = await fetch(`${this.baseUrl}/api/pull`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ name: modelName })
            });

            return await response.json();
        } catch (error) {
            console.error('Erro ao baixar modelo:', error);
            throw error;
        }
    }
}

// Exemplo de uso
async function main() {
    const client = new OllamaClient();

    try {
        // 1. Listar modelos disponíveis
        console.log('📋 Modelos disponíveis:');
        const models = await client.listModels();
        models.models?.forEach(model => {
            console.log(`  - ${model.name}`);
        });

        // 2. Gerar resposta simples
        console.log('\n🤖 Gerando resposta...');
        const response = await client.generate(
            'Explique o que é machine learning em 2 frases simples',
            'llama2'
        );
        console.log(`Resposta: ${response.response || 'Erro na resposta'}`);

        // 3. Chat com histórico
        console.log('\n💬 Chat com histórico:');
        const messages = [
            { role: 'user', content: 'Olá! Como você pode me ajudar?' },
            { role: 'assistant', content: 'Olá! Sou um assistente de IA e posso ajudar com várias tarefas.' },
            { role: 'user', content: 'Pode me dar 3 dicas para aprender JavaScript?' }
        ];

        const chatResponse = await client.chat(messages, 'llama2');
        console.log(`Resposta do chat: ${chatResponse.message?.content || 'Erro'}`);

    } catch (error) {
        console.error('Erro na execução:', error);
    }
}

// Executar se for o arquivo principal
if (typeof require !== 'undefined' && require.main === module) {
    main();
}

// Exportar para uso em módulos
if (typeof module !== 'undefined' && module.exports) {
    module.exports = OllamaClient;
} 

File: examples/python/ollama_client.py
#!/usr/bin/env python3
"""
Cliente Python para API do Ollama
Exemplo prático de integração com IA local
"""

import requests
import json
import time
from typing import Dict, Any, Optional

class OllamaClient:
    def __init__(self, base_url: str = "http://localhost:11434"):
        self.base_url = base_url
        self.session = requests.Session()
    
    def list_models(self) -> Dict[str, Any]:
        """Lista todos os modelos disponíveis"""
        response = self.session.get(f"{self.base_url}/api/tags")
        return response.json()
    
    def generate(self, prompt: str, model: str = "llama2", 
                stream: bool = False, **kwargs) -> Dict[str, Any]:
        """Gera resposta para um prompt"""
        data = {
            "model": model,
            "prompt": prompt,
            "stream": stream,
            **kwargs
        }
        
        response = self.session.post(f"{self.base_url}/api/generate", json=data)
        return response.json()
    
    def chat(self, messages: list, model: str = "llama2") -> Dict[str, Any]:
        """Chat com histórico de mensagens"""
        data = {
            "model": model,
            "messages": messages
        }
        
        response = self.session.post(f"{self.base_url}/api/chat", json=data)
        return response.json()
    
    def pull_model(self, model_name: str) -> Dict[str, Any]:
        """Baixa um modelo específico"""
        data = {"name": model_name}
        response = self.session.post(f"{self.base_url}/api/pull", json=data)
        return response.json()

def main():
    """Exemplo de uso do cliente"""
    client = OllamaClient()
    
    # 1. Listar modelos disponíveis
    print("📋 Modelos disponíveis:")
    models = client.list_models()
    for model in models.get('models', []):
        print(f"  - {model['name']}")
    
    # 2. Gerar resposta simples
    print("\n🤖 Gerando resposta...")
    response = client.generate(
        prompt="Explique o que é machine learning em 2 frases simples",
        model="llama2"
    )
    print(f"Resposta: {response.get('response', 'Erro na resposta')}")
    
    # 3. Chat com histórico
    print("\n💬 Chat com histórico:")
    messages = [
        {"role": "user", "content": "Olá! Como você pode me ajudar?"},
        {"role": "assistant", "content": "Olá! Sou um assistente de IA e posso ajudar com várias tarefas."},
        {"role": "user", "content": "Pode me dar 3 dicas para aprender Python?"}
    ]
    
    chat_response = client.chat(messages, model="llama2")
    print(f"Resposta do chat: {chat_response.get('message', {}).get('content', 'Erro')}")

if __name__ == "__main__":
    main() 